feat(versioning/deb): New module to compare deb package versions like dpkg (#20291)
renovatebot/renovate: lib/modules/versioning/deb/readme.md

> "deb" versioning compares versions like package managers on Debian based Linux distributions compare packages (dpkg, apt).

Reviewer: ```suggestion
Deb versioning compares versions like package managers on Debian-based Linux distributions compare packages (`dpkg`, `apt`).
```

> They support rather complicated version numbers (including alphabetical characters almost everywhere).

Reviewer: ```suggestion
Deb versioning supports complicated version numbers, including alphabetical characters almost everywhere.
```

> Just like Renovate's "loose" versioning it tries its best to extract found numbers to fill SemVer known fields like major, minor, patch.

Reviewer: ```suggestion
Similar to our "loose" versioning, deb versioning is a "best effort" attempt to convert deb versions to SemVer fields like `major`, `minor`, `patch`.
```
